Compile failure on php 7.4

After installing php 7.4 and run:

```
pecl install amqp
```

I got some compilation errors.

```
running: make
/bin/sh /private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/libtool --mode=compile cc  -I. -I/private/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/include -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/main -I/private/tmp/pear/temp/amqp -I/usr/local/Cellar/php/7.4.0/include/php -I/usr/local/Cellar/php/7.4.0/include/php/main -I/usr/local/Cellar/php/7.4.0/include/php/TSRM -I/usr/local/Cellar/php/7.4.0/include/php/Zend -I/usr/local/Cellar/php/7.4.0/include/php/ext -I/usr/local/Cellar/php/7.4.0/include/php/ext/date/lib -I/usr/local/Cellar/rabbitmq-c/0.9.0_1/include  -DHAVE_CONFIG_H  -g -O2   -c /private/tmp/pear/temp/amqp/amqp.c -o amqp.lo
mkdir .libs
 cc -I. -I/private/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/include -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/main -I/private/tmp/pear/temp/amqp -I/usr/local/Cellar/php/7.4.0/include/php -I/usr/local/Cellar/php/7.4.0/include/php/main -I/usr/local/Cellar/php/7.4.0/include/php/TSRM -I/usr/local/Cellar/php/7.4.0/include/php/Zend -I/usr/local/Cellar/php/7.4.0/include/php/ext -I/usr/local/Cellar/php/7.4.0/include/php/ext/date/lib -I/usr/local/Cellar/rabbitmq-c/0.9.0_1/include -DHAVE_CONFIG_H -g -O2 -c /private/tmp/pear/temp/amqp/amqp.c  -fno-common -DPIC -o .libs/amqp.o
/bin/sh /private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/libtool --mode=compile cc  -I. -I/private/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/include -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/main -I/private/tmp/pear/temp/amqp -I/usr/local/Cellar/php/7.4.0/include/php -I/usr/local/Cellar/php/7.4.0/include/php/main -I/usr/local/Cellar/php/7.4.0/include/php/TSRM -I/usr/local/Cellar/php/7.4.0/include/php/Zend -I/usr/local/Cellar/php/7.4.0/include/php/ext -I/usr/local/Cellar/php/7.4.0/include/php/ext/date/lib -I/usr/local/Cellar/rabbitmq-c/0.9.0_1/include  -DHAVE_CONFIG_H  -g -O2   -c /private/tmp/pear/temp/amqp/amqp_type.c -o amqp_type.lo
 cc -I. -I/private/tmp/pear/temp/amqp -DPHP_ATOM_INC -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/include -I/private/tmp/pear/temp/pear-build-hugoqKJyG7/amqp-1.9.4/main -I/private/tmp/pear/temp/amqp -I/usr/local/Cellar/php/7.4.0/include/php -I/usr/local/Cellar/php/7.4.0/include/php/main -I/usr/local/Cellar/php/7.4.0/include/php/TSRM -I/usr/local/Cellar/php/7.4.0/include/php/Zend -I/usr/local/Cellar/php/7.4.0/include/php/ext -I/usr/local/Cellar/php/7.4.0/include/php/ext/date/lib -I/usr/local/Cellar/rabbitmq-c/0.9.0_1/include -DHAVE_CONFIG_H -g -O2 -c /private/tmp/pear/temp/amqp/amqp_type.c  -fno-common -DPIC -o .libs/amqp_type.o
/private/tmp/pear/temp/amqp/amqp_type.c:87:7: error: expected ';' after expression
        ulong index;
             ^
             ;
/private/tmp/pear/temp/amqp/amqp_type.c:87:2: error: use of undeclared identifier 'ulong'
        ulong index;
        ^
/private/tmp/pear/temp/amqp/amqp_type.c:92:2: error: non-object type 'char *(const char *, int)' is not assignable
        PHP5to7_ZEND_HASH_FOREACH_KEY_VAL(ht, index, real_key, key, key_len, data, value, pos) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/pear/temp/amqp/php7_support.h:78:3: note: expanded from macro 'PHP5to7_ZEND_HASH_FOREACH_KEY_VAL'
  ZEND_HASH_FOREACH_KEY_VAL((ht), (num_key), (real_key), (val))
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/php/7.4.0/include/php/Zend/zend_hash.h:1005:5: note: expanded from macro 'ZEND_HASH_FOREACH_KEY_VAL'
        _h = _p->h; \
        ~~ ^
/private/tmp/pear/temp/amqp/amqp_type.c:87:8: warning: expression result unused [-Wunused-value]
        ulong index;
              ^~~~~
/private/tmp/pear/temp/amqp/amqp_type.c:122:7: error: expected ';' after expression
        ulong index;
             ^
             ;
/private/tmp/pear/temp/amqp/amqp_type.c:122:2: error: use of undeclared identifier 'ulong'
        ulong index;
        ^
/private/tmp/pear/temp/amqp/amqp_type.c:130:2: error: non-object type 'char *(const char *, int)' is not assignable
        PHP5to7_ZEND_HASH_FOREACH_KEY_VAL(ht, index, real_key, key, key_len, data, value, pos) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/pear/temp/amqp/php7_support.h:78:3: note: expanded from macro 'PHP5to7_ZEND_HASH_FOREACH_KEY_VAL'
  ZEND_HASH_FOREACH_KEY_VAL((ht), (num_key), (real_key), (val))
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/php/7.4.0/include/php/Zend/zend_hash.h:1005:5: note: expanded from macro 'ZEND_HASH_FOREACH_KEY_VAL'
        _h = _p->h; \
        ~~ ^
/private/tmp/pear/temp/amqp/amqp_type.c:143:35: warning: format specifies type 'unsigned long' but the argument has type 'char *(*)(const char *, int)' [-Wformat]
                                key_len = sprintf(str, "%lu", index);
                                                        ~~~   ^~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note: expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
/private/tmp/pear/temp/amqp/amqp_type.c:147:91: warning: format specifies type 'unsigned long' but the argument has type 'char *(*)(const char *, int)' [-Wformat]
                                php_error_docref(NULL TSRMLS_CC, E_WARNING, "Ignoring non-string header field '%lu'", index);
                                                                                                               ~~~    ^~~~~
/private/tmp/pear/temp/amqp/amqp_type.c:122:8: warning: expression result unused [-Wunused-value]
        ulong index;
              ^~~~~
/private/tmp/pear/temp/amqp/amqp_type.c:185:7: error: expected ';' after expression
        ulong index;
             ^
             ;
/private/tmp/pear/temp/amqp/amqp_type.c:185:2: error: use of undeclared identifier 'ulong'
        ulong index;
        ^
/private/tmp/pear/temp/amqp/amqp_type.c:195:2: error: non-object type 'char *(const char *, int)' is not assignable
        PHP5to7_ZEND_HASH_FOREACH_KEY_VAL(ht, index, real_key, key, key_len, data, value, pos) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/private/tmp/pear/temp/amqp/php7_support.h:78:3: note: expanded from macro 'PHP5to7_ZEND_HASH_FOREACH_KEY_VAL'
  ZEND_HASH_FOREACH_KEY_VAL((ht), (num_key), (real_key), (val))
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/Cellar/php/7.4.0/include/php/Zend/zend_hash.h:1005:5: note: expanded from macro 'ZEND_HASH_FOREACH_KEY_VAL'
        _h = _p->h; \
        ~~ ^
/private/tmp/pear/temp/amqp/amqp_type.c:185:8: warning: expression result unused [-Wunused-value]
        ulong index;
              ^~~~~
5 warnings and 9 errors generated.
make: *** [amqp_type.lo] Error 1
ERROR: `make' failed
```

--- a/amqp_basic_properties.c
+++ b/amqp_basic_properties.c
@@ -424,7 +424,7 @@ void parse_amqp_table(amqp_table_t *table, zval *result TSRMLS_DC) {
                             PHP5to7_ADD_NEXT_INDEX_STRINGL_DUP(
                                     PHP5to7_MAYBE_PTR(value),
                                     entry->value.value.array.entries[j].value.bytes.bytes,
-                                    (uint) entry->value.value.array.entries[j].value.bytes.len
+                                    (unsigned) entry->value.value.array.entries[j].value.bytes.len
                             );
                             break;
                         case AMQP_FIELD_KIND_TABLE: {
@@ -507,7 +507,7 @@ void parse_amqp_table(amqp_table_t *table, zval *result TSRMLS_DC) {
         }
 
         if (has_value) {
-            char *key = estrndup(entry->key.bytes, (uint) entry->key.len);
+            char *key = estrndup(entry->key.bytes, (unsigned) entry->key.len);
             add_assoc_zval(result, key, PHP5to7_MAYBE_PTR(value));
             efree(key);
         } else {

--- a/amqp_exchange.c
+++ b/amqp_exchange.c
@@ -244,7 +244,7 @@ static PHP_METHOD(amqp_exchange_class, hasArgument)
 		return;
 	}
 
-	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint)(key_len + 1), tmp)) {
+	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned)(key_len + 1), tmp)) {
 		RETURN_FALSE;
 	}
 
@@ -296,13 +296,13 @@ static PHP_METHOD(amqp_exchange_class, setArgument)
 
 	switch (Z_TYPE_P(value)) {
 		case IS_NULL:
-			PHP5to7_ZEND_HASH_DEL(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint) (key_len + 1));
+			PHP5to7_ZEND_HASH_DEL(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned) (key_len + 1));
 			break;
 		PHP5to7_CASE_IS_BOOL:
 		case IS_LONG:
 		case IS_DOUBLE:
 		case IS_STRING:
-			PHP5to7_ZEND_HASH_ADD(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint) (key_len + 1), value, sizeof(zval *));
+			PHP5to7_ZEND_HASH_ADD(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned) (key_len + 1), value, sizeof(zval *));
 			Z_TRY_ADDREF_P(value);
 			break;
 		default:

--- a/amqp_queue.c
+++ b/amqp_queue.c
@@ -200,7 +200,7 @@ static PHP_METHOD(amqp_queue_class, getArgument)
 		return;
 	}
 
-	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint)(key_len + 1), tmp)) {
+	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned)(key_len + 1), tmp)) {
 		RETURN_FALSE;
 	}
 
@@ -221,7 +221,7 @@ static PHP_METHOD(amqp_queue_class, hasArgument)
 		return;
 	}
 
-	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint)(key_len + 1), tmp)) {
+	if (!PHP5to7_ZEND_HASH_FIND(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned)(key_len + 1), tmp)) {
 		RETURN_FALSE;
 	}
 
@@ -274,13 +274,13 @@ static PHP_METHOD(amqp_queue_class, setArgument)
 
 	switch (Z_TYPE_P(value)) {
 		case IS_NULL:
-			PHP5to7_ZEND_HASH_DEL(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint) (key_len + 1));
+			PHP5to7_ZEND_HASH_DEL(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned) (key_len + 1));
 			break;
 		PHP5to7_CASE_IS_BOOL:
 		case IS_LONG:
 		case IS_DOUBLE:
 		case IS_STRING:
-			PHP5to7_ZEND_HASH_ADD(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (uint) (key_len + 1), value, sizeof(zval *));
+			PHP5to7_ZEND_HASH_ADD(PHP_AMQP_READ_THIS_PROP_ARR("arguments"), key, (unsigned) (key_len + 1), value, sizeof(zval *));
 			Z_TRY_ADDREF_P(value);
 			break;
 		default:
@@ -565,7 +565,7 @@ static PHP_METHOD(amqp_queue_class, consume)
 		}
 
 		char *key;
-		key = estrndup((char *) r->consumer_tag.bytes, (uint) r->consumer_tag.len);
+		key = estrndup((char *) r->consumer_tag.bytes, (unsigned) r->consumer_tag.len);
 
 		if (PHP5to7_ZEND_HASH_FIND(Z_ARRVAL_P(consumers), (const char *) key, PHP5to7_ZEND_HASH_STRLEN(r->consumer_tag.len), consumer_tag_zv)) {
 			// This should never happen as AMQP server guarantees that consumer tag is unique within channel
@@ -683,7 +683,7 @@ static PHP_METHOD(amqp_queue_class, consume)
 		}
 
 		char *key;
-		key = estrndup((char *)envelope.consumer_tag.bytes, (uint) envelope.consumer_tag.len);
+		key = estrndup((char *)envelope.consumer_tag.bytes, (unsigned) envelope.consumer_tag.len);
 
 		if (!PHP5to7_ZEND_HASH_FIND(Z_ARRVAL_P(consumers), key, PHP5to7_ZEND_HASH_STRLEN(envelope.consumer_tag.len), current_queue_zv)) {
 			PHP5to7_zval_t exception PHP5to7_MAYBE_SET_TO_NULL;
@@ -982,7 +982,7 @@ static PHP_METHOD(amqp_queue_class, cancel)
 	}
 
     char *key;
-    key = estrndup((char *)r->consumer_tag.bytes, (uint) r->consumer_tag.len);
+    key = estrndup((char *)r->consumer_tag.bytes, (unsigned) r->consumer_tag.len);
     PHP5to7_ZEND_HASH_DEL(Z_ARRVAL_P(consumers), (const char *) key, PHP5to7_ZEND_HASH_STRLEN(r->consumer_tag.len));
     efree(key);
 

--- a/amqp_type.c
+++ b/amqp_type.c
@@ -82,9 +82,9 @@ void php_amqp_type_internal_convert_zval_array(zval *php_array, amqp_field_value
 	PHP5to7_ZEND_REAL_HASH_KEY_T *real_key;
 
 	char *key;
-	uint key_len;
+	unsigned key_len;
 
-	ulong index;
+	zend_ulong index;
 	ht = Z_ARRVAL_P(php_array);
 
 	zend_bool is_amqp_array = 1;
@@ -117,9 +117,9 @@ void php_amqp_type_internal_convert_zval_to_amqp_table(zval *php_array, amqp_tab
 	PHP5to7_ZEND_REAL_HASH_KEY_T *real_key;
 
 	char *key;
-	uint key_len;
+	unsigned key_len;
 
-	ulong index;
+	zend_ulong index;
 
 
 	ht = Z_ARRVAL_P(php_array);
@@ -180,9 +180,9 @@ void php_amqp_type_internal_convert_zval_to_amqp_array(zval *zvalArguments, amqp
 	PHP5to7_ZEND_REAL_HASH_KEY_T *real_key;
 
 	char *key;
-	uint key_len;
+	unsigned key_len;
 
-	ulong index;
+	zend_ulong index;
 
 	char type[16];
 
@@ -232,7 +232,7 @@ zend_bool php_amqp_type_internal_convert_php_to_amqp_field_value(zval *value, am
 			if (Z_STRLEN_P(value)) {
 				amqp_bytes_t bytes;
 				bytes.len = (size_t) Z_STRLEN_P(value);
-				bytes.bytes = estrndup(Z_STRVAL_P(value), (uint) Z_STRLEN_P(value));
+				bytes.bytes = estrndup(Z_STRVAL_P(value), (unsigned) Z_STRLEN_P(value));
 
 				field->value.bytes = bytes;
 			} else {

--- a/php5_support.h
+++ b/php5_support.h
@@ -42,14 +42,14 @@ typedef zval* PHP5to7_zval_t;
 
 #define PHP5to7_ZVAL_STRINGL_DUP(z, s, l) ZVAL_STRINGL((z), (s), (l), 1)
 
-#define PHP5to7_ADD_NEXT_INDEX_STRINGL_DUP(arg, str, length) add_next_index_stringl((arg), (str), (uint)(length), 1)
+#define PHP5to7_ADD_NEXT_INDEX_STRINGL_DUP(arg, str, length) add_next_index_stringl((arg), (str), (unsigned)(length), 1)
 
 #define PHP5to7_ZEND_HASH_FIND(ht, str, len, res) \
-        (zend_hash_find((ht), (str), (uint)(len), (void **) &(res)) != FAILURE)
+        (zend_hash_find((ht), (str), (unsigned)(len), (void **) &(res)) != FAILURE)
 
-#define PHP5to7_ZEND_HASH_STRLEN(len) (uint)((len) + 1)
-#define PHP5to7_ZEND_HASH_DEL(ht, key, len) zend_hash_del_key_or_index((ht), (key), (uint)(len), 0, HASH_DEL_KEY);
-#define PHP5to7_ZEND_HASH_ADD(ht, key, len, pData, nDataSize) (zend_hash_add((ht), (key), (uint)(len), &(pData), nDataSize, NULL) != FAILURE)
+#define PHP5to7_ZEND_HASH_STRLEN(len) (unsigned)((len) + 1)
+#define PHP5to7_ZEND_HASH_DEL(ht, key, len) zend_hash_del_key_or_index((ht), (key), (unsigned)(len), 0, HASH_DEL_KEY);
+#define PHP5to7_ZEND_HASH_ADD(ht, key, len, pData, nDataSize) (zend_hash_add((ht), (key), (unsigned)(len), &(pData), nDataSize, NULL) != FAILURE)
 #define PHP5to7_ZEND_HASH_STR_UPD_MEM(ht, key, len, pData, nDataSize) PHP5to7_ZEND_HASH_ADD((ht), (key), (len), (pData), (nDataSize))
 #define PHP5to7_ZEND_HASH_STR_FIND_PTR(ht, key, len, res) PHP5to7_ZEND_HASH_FIND((ht), (key), (len), (res))
 #define PHP5to7_ZEND_HASH_STR_DEL(ht, key, len) PHP5to7_ZEND_HASH_DEL((ht), (key), (len))

--- a/php7_support.h
+++ b/php7_support.h
@@ -47,11 +47,11 @@ typedef zval PHP5to7_zval_t;
 		((res = zend_hash_str_find((ht), (str), (size_t)(len - 1))) != NULL)
 
 #define PHP5to7_ZEND_HASH_STRLEN(len) (PHP5to7_param_str_len_type_t)((len) + 1)
-#define PHP5to7_ZEND_HASH_DEL(ht, key, len) zend_hash_str_del_ind((ht), (key), (uint)(len - 1))
-#define PHP5to7_ZEND_HASH_ADD(ht, key, len, pData, nDataSize) zend_hash_str_add((ht), (key), (uint)(len - 1), (pData))
+#define PHP5to7_ZEND_HASH_DEL(ht, key, len) zend_hash_str_del_ind((ht), (key), (unsigned)(len - 1))
+#define PHP5to7_ZEND_HASH_ADD(ht, key, len, pData, nDataSize) zend_hash_str_add((ht), (key), (unsigned)(len - 1), (pData))
 #define PHP5to7_ZEND_HASH_STR_UPD_MEM(ht, key, len, pData, nDataSize) zend_hash_str_update_mem((ht), (key), (size_t)(len), &(pData), (nDataSize))
 #define PHP5to7_ZEND_HASH_STR_FIND_PTR(ht, key, len, res) ((res = zend_hash_str_find_ptr((ht), (key), (size_t)(len))) != NULL)
-#define PHP5to7_ZEND_HASH_STR_DEL(ht, key, len) zend_hash_str_del_ind((ht), (key), (uint)(len))
+#define PHP5to7_ZEND_HASH_STR_DEL(ht, key, len) zend_hash_str_del_ind((ht), (key), (unsigned)(len))
 
 #define PHP5to7_SET_FCI_RETVAL_PTR(fci, pzv) (fci).retval = (pzv);
 #define PHP5to7_CHECK_FCI_RETVAL_PTR(fci) ((fci).retval)

